packages/fxa-payments-server/src/routes/Checkout/index.tsx

```diff
@@ -178,8 +178,10 @@ export const Checkout = ({
           retryStatus,
           onSuccess: fetchProfileAndCustomer,
           onFailure: setSubscriptionError,
-          onRetry: (status: RetryStatus) => {
-            setRetryStatus(status);
+          onRetry: () => {
+            // for a passwordless subscription we retry from a clean state
+            // every time.  new stub account, new customer, new invoice, etc.
+            setRetryStatus(undefined);
```

Comment:
`onRetry` is a callback that's passed multiple levels down and then simply called.  But, since there's no logic in the function, I don't think it's worth the amount of mocking (and likely refactoring) involved to truly test it.  The only way this stops calling `setRetryStatus` with the _hard coded_ `undefined` is someone make an edit in the code to change the argument.

I will file an issue for a functional test for a failed-then-successful payment subscription though.

Reply:
Might be good to raise in @ashrivastava-qa 's SubPlat functional testing doc that she shared in a team meeting a few weeks ago?

Reply:
> Might be good to raise in @ashrivastava-qa 's SubPlat functional testing doc that she shared in a team meeting a few weeks ago?

Good idea.  Thanks for the reminder.